04-onion-routing.md

```diff
@@ -367,6 +368,102 @@ otherwise meets the amount criterion (eg. some other failure, or
 invoice timeout), however if it were to fulfill only some of them,
 intermediary nodes could simply claim the remaining ones.
 
+### Onion Messages
+
+Onion messages have an onion with an alternate `hop_payload`
+format: a `bigsize` followed by a `onionmsg_payload`.  Note that there
+is no legacy format, thus a `bigsize` of 0 means no payload.
+
+An intermediate node expects an `enctlv` which it can decrypt using
+the `blinding` which it is handed along with the onion message.
+
+1. `tlv_stream`: `onionmsg_payload`
+2. types:
+    1. type: 8 (`reply_path`)
+    2. data:
+        * [`point`:`first_node_id`]
+        * [`point`:`blinding`]
+        * [`...*onionmsg_path`:`path`]
+    1. type: 10 (`enctlv`)
+    2. data:
+        * [`...*byte`:`enctlv`]
+
+1. `tlv_stream`: `encmsg_tlvs`
+2. types:
+    1. type: 1 (`padding`)
+    2. data:
+        * [`...*byte`:`pad`]
+    1. type: 4 (`next_node_id`)
+    2. data:
+        * [`point`:`node_id`]
+    1. type: 12 (`next_blinding`)
+    2. data:
+        * [`point`:`blinding`]
+
+1. subtype: `onionmsg_path`
+2. data:
+    * [`point`:`node_id`]
+    * [`u16`:`enclen`]
+    * [`enclen*byte`:`enctlv`]
```

Comment:
It is not specified at all what this is, nor does route blinding tell you what it is, as far as I can tell. The route blinding doc does not either, or at least if it does you don't specify where to look for it.

Reply:
Actually now that I realize that onion messages uses route blinding on the way forward as well, it's not encrypted by the recipient but by the sender (but on the reply path it will be encrypted by the recipient), so my name sucks...

I cannot do better than `encrypted_data` then, wdyt?